### ISSUE TITLE
Tiberium Poisoning Improvements

### ISSUE THREAD

**Issue by RealTelefonmast** [Profugo-Barbatus/TiberiumRim#72]
Player suggestion:
"Frostbite might serve as a good base for tiberium poisoning. After the player gets a certain value for "cold" the frostbite will be applied. If the players stays cold, it will be applied in more places and worsen. Using this as base for the tiberium poisoning, the player will get a 'tiberium exposure' de-buff when walking through (or maybe near) tiberium fields. This will increase rapidly unless the player is clothed in protective gear. Moving to a safer place will really slowly reduce the de-buff.

When the tiberium exposure reaches a certain level, 'Tiberium Toxemia' may happen. This can happen as Poisoning, Assimilation or Mutation.

Tiberium poisoning would cause all kinds of randomized bad effects like reduced metabolism, reduced blood pumping/filtration and other negative effects, and would last really long. (these are the effects of low level tiberium exposure according to the cnc wikia).

Tiberium assimilation is when tiberium starts growing on, or in, your body. Its fatal really quickly, and mostly the same as the current 'natural tiberium poisoning'. It can happen by breathing in small tiberium particles, falling over and touching a piece of tiberium, or cutting yourself on a sharp edge of a tiberium crystal, leaving the particles in the wound. The current only cure is amputation.

If a mutation happens during the assimilation process, the host will adapt to the tiberium.
my suggestions for mutants are;
-Extremely low chance of occurring.
-Immune to tiberium exposure and poisoning. (or even dependant/addicted to tiberium radiation)
-Assimilation won't be fatal, and there might even be positive effects. (maybe some positive and negative ones so it can't be abused)
-Immune to further mutation?"

**Comment by Nephlite** [comment]
I agree with this opinion. pawns die too quickly and easily by Tiberium addiction. We need to make a little more sophisticated improvements.

**Comment by Profugo-Barbatus** [comment]
I agree that granularity with the Tiberium Infection system would be good. I might not go so far as to include it with 1.3, but exact release date is minor, I'd just not advise holding up 1.3 for this component. Especially since this would make save games incompatible. My recommendation would be to finish this, in isolation, and not merge the branch until the release of A17, which would mandate new saves anyhow. Easiest solution on that front.

Frostbite is actually a terrible basis for this, as it is incredibly hard coded when it comes to these minute environmental exposures. It was something I looked into back when I first wrote up some infection mechanics.

Shifting the specific Tiberium infections from guaranteed lethality to a gradual system would be an acceptable compromise. Staying within the realm of the reasonably coded, My proposal for a granulated system based on your desires would be as follows. Keep in mind, The randomly branching progression isn't included in this, as players would generally prefer a predictable approach to treatment and lethality with such a common material. Prisoners from raids are usually infested with Tiberium, and so being able to predict the effects makes it easier to decide who's worth saving.

Stage 1: Tiberium remains a localized "infection" type system, but rather than irreversibly ticking up to death, stage 1 would be a treatable disease, would not tick up when untreated (Or would only tick up slowly), and would very slowly tick down. While not lethal, stage one should retain some serious debuffs to reflect the canon of Tiberium exposure, even limited, being exceedingly bad. If the exposure advances far enough, then it moves to a stage 2 disease. The exact rate of increase and decrease would be tweak able for balance, but the general idea is that Tiberium exposure should mandate a few days rest for a pawn. Consider this to essentially be the "Tiberium Poisoning" stage.

Stage 2: Once the infection has reached stage 2, it is Permanent, as its begun the assimilation process (A faction-pack specific expensive medicine could be introduced at a later point, as an option from one of the tech tree's that reverts Stage 2 infections back to stage 1). The infection will remain treatable, but treatment will simply prevent the disease from advancing, and will not reduce it. Consider this the Tiberium Assimilation. It goes against lore a bit, but allowing treatment to prevent advancing of the disease would give players a chance to prepare whatever they needed for the traditional amputation approach, or to acquire any potential treatments that are available down the road. Think of it as "Intensive surgery to reduce internal growth". Whenever the infection is not treated, and ticking up, we could consider having it essentially 'roll' against slim odds to either gain a body-wide Tiberium Immunity effect (Untreatable, eternal body-wide hediff that all Tiberium Infection checks against) that prevents any further infections. Variations of this could be introduced over time as different forms of mutation without much additional difficulty, to allow both good and bad mutation to come alongside the immunity. Stuff such as debuffs to movement if they're unlucky (Mutated legs and shit), or buffs to manipulation or health related matters (If they're lucky). Again, with faction pack integration, we could look into allowing late/end game drugs to increase the odds of mutation, and increase the odds of specific mutations for NOD's super soldier programs.

The reasoning for all the above is that it fits well within the accepted Rimworld progression of sickness (We can even name the first few stages of Tiberium Poisoning to be Tiberium exposure), Fits reasonably well in the lore, addresses the players concerns while retaining lethality in the long run, adds options we can hook the faction release onto later, and doesn't require us to invent entire new hediff classes or overwrite base function.

**Comment by Nephlite** [comment]
If possible, the disease part is also recommended to be modified together in version 1.3.

Currently, save files in versions 1.2 and 1.3 of Tiberium Rim are incompatible. Many things have already changed. It is better to fix them at once in version 1.3.

I remember that Hugslib had an update note messaging feature. If possible, release minor update 1.2.1 first to warn of the destruction of the save file. I think no one will complain if we warn specific timings for 1.3 updates in Steam, Forums, and Ingame messages.

**Comment by RealTelefonmast** [comment]
Only now realized the response.

Alright, I need to agree with Neph that sadly incompatibility with current saves seems to have already been reached. 
About your ideas on the infections, they are just the way I had in mind. What I posted as the issue was merely the a quote of the player suggestion, I myself have seen that frostbite is not a usable base for this. Instead i thought of different hediffs that occur on different stages. 
Tiberium exposure is the base of everything, it ticks up as long as a living being is on tiberium, affects whole body.
Stage one affects an outer body part, in the same manner as you explained, sets in after a certain amount of exposure.
Stage two affects the same body part, permanently "ruins" it. If lucky as you said, mutation may set in, allowing all kinds of mutations, from unlucky diformations to enhancements or "tiberium need" which is satisfied by sitting in a field and so on. Depending on agreement of what types of mutations should exist, while these three seem to serve a good base that might be enough. 
Then corresponding to the level of exposure the different stage hediffs set in, exposure will possibly remain as a way to measure your pawns situation, might be changed if i find a better solution while coding.

So once you give me the greenlight on this I will start implementing the new infection mechanics, possibly as a seperate branch even tho the incompatibility is already present, needs to be tested on my side though, will give feedback of results once available.
